docs/jd-delivery-api-decision-adapter.md

```diff
@@ -0,0 +1,112 @@
+---
+id: js-delivery-api-decision-adapter
+title: Customise your decision adapter & bucketing
+sidebar_label: Customise decsion adapter
+---
+
+Mojito gives users control over the decision/bucketing engine that assigns subjects to tests and recipes. This is especially useful in an ITP 2.3 world devoid of lasting client-set cookies.
+
+The decision adapter lets you:
+
+ - Bucket subjects consistently across devices (e.g. assigning tests by a customer ID)
+ - Work around ITP 2.3 restrictions (e.g. by using a server-set cookie ID)
+ - Implement your own, preferred PRNG or hash function
+
+It gives you unprecedented control over how subjects are assigned.
+
+## Mojito.options.decisionAdapter
+
+When subjects are bucketed and assigned a recipe ([see `Assign subject by sample rate` on the execution flowchart](js-delivery-api-execution-order#split-test-object-execution-flowchart)), the decisionAdapter is called. This can happen up to three times per experiment ([as described below](#expect-to-handle-multiple-decisions-per-test)).
+
+### Syntax
+
+`Mojito.options.decisionAdapter = function (test) {...};`
+
+Parameter | &nbsp;
+--|--
+**test** <br> Type: *object* <br> *Required* | The fully constructed test object is passed into this function in the first position. You can access everything within the test object or store values behind new/existing keys for later use.
+
+Return value | &nbsp;
+--|--
+Type: *float* | The decision returned (must be between 0 and 1)
+
+
+## Example: Set a new default decision adapter
+
+Inside your `./lib/shared-code.js` you could define a container-wide adapter like so:
+
+```js
+/**
+ * My Custom Decision Adapter
+ * @param {Object} test: The full test object to access within the decisionAdapter
+ * @returns {Float}: The decision - a number between 0 and 1
+ */
+Mojito.options.decisionAdapter = function (test)
+{
+    return Math.random();
+};
+```
+
+It must return a value between 0 and 1. E.g. Consider an experiment with a 50-50 split between `Control` and `Treatment` groups - if the decision adapter returns a value of `0.314...`, then the user would be assigned the `Control`. Whereas a value of `0.694...` would bucket them into the `Treatment` group.
```

Comment:
Maybe a simpler float could convey the point clearer. E.g. `0.3` / `0.7`

Reply:
Yeah, true...